### PR TITLE
XWIKI-17654: Errors related to user mentions still displayed in conso…

### PR DESCRIPTION
**Pull request by manuelleduc**
…le when upgrading XWiki

https://jira.xwiki.org/browse/XWIKI-17654

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionsDataConsumer.java [hunk 1]
```diff
@@ -130,26 +130,29 @@ public void consume(MentionsData data) throws XWikiException
...
+                XWikiDocument oldDoc;
+                if (doc.getPreviousVersion() != null) {
+                    oldDoc = this.documentRevisionProvider.getRevision(dr, doc.getPreviousVersion());
+                } else {
+                    oldDoc = null;
+                }
+                DocumentReference documentReference = doc.getDocumentReference();
+                DocumentReference authorReference = doc.getAuthorReference();
+
+                if (oldDoc == null) {
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionsDataConsumer.java:144]: this if should be mergeable with the previous IF, no?

File: xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionsDataConsumer.java [hunk 2]
```diff
@@ -185,7 +188,9 @@ private void traverseXObjectsOnUpdate(Map<DocumentReference, List<BaseObject>> o
         for (Map.Entry<DocumentReference, List<BaseObject>> entry : xObjects.entrySet()) {
             List<BaseObject> oldEntry = oldXObjects.get(entry.getKey());
             for (BaseObject baseObject : entry.getValue()) {
-                handBaseObject(oldEntry, baseObject, documentReference, authorReference, syntax);
+                if (baseObject != null) {
+                    handBaseObject(oldEntry, baseObject, documentReference, authorReference, syntax);
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionsDataConsumer.java:192]: typo? hand vs handle?

File: xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionsDataConsumer.java (hunk 1 above)
Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionsDataConsumer.java]: It is. I've simplified the method.

File: xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionsDataConsumer.java (hunk 2 above)
Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionsDataConsumer.java]: Well spotted. Thanks.

**Comment by surli** [comment]
I guess you could provide a unit test for that one?

File: xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/test/java/org/xwiki/mentions/internal/DefaultMentionsDataConsumerTest.java [hunk 3]
```diff
@@ -320,6 +320,150 @@ void consumeUpdate() throws Exception
                 newCommentXDOM));
     }
 
+    @Test
+    void consumeUpdateBaseObjectNull() throws Exception
```
Comment (surli) [inline xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/test/java/org/xwiki/mentions/internal/DefaultMentionsDataConsumerTest.java:324]: Those tests are difficult to read: couldn't you factorize some piece of code for the fixture of the test? I think most of the content of the test is similar from one test to another no?

**Pull request merged**